client/src/components/Navbar/components/Drawer/index.tsx

```diff
@@ -0,0 +1,234 @@
+  const [port, setPort] = React.useState("");
+  const [login, setLogin] = React.useState("");
+  const [password, setPassword] = React.useState("");
+  const [database, setDatabase] = React.useState("");
+  const [type, setType] = React.useState(types[0]);
+  const [hasChanged, setHasChanged] = React.useState(false);
+  const [hasSuccessfullyChanged, setHasSuccessfullyChanged] = React.useState(
+    false
+  );
+  const [loading, setLoading] = React.useState(true);
+  const selectedNode = useSelector((state: IReduxStore) => state.selectedNode);
+  const client = useApolloClient();
+
+  React.useEffect(() => {
+    console.log("USE EFFECT", selectedNode);
+    if (selectedNode.source.id) {
+      client
+        .query({
+          query: credential,
+          variables: {
+            sourceId: selectedNode.source.id
+          },
+          fetchPolicy: "network-only"
+        })
+        .then((response: any) => {
+          setLoading(false);
+
+          const cred = response.data.credential;
+          setHost(cred.host);
+          setPort(cred.port);
+          setLogin(cred.login);
+          setPassword(cred.password);
+          setDatabase(cred.database);
+          setType(cred.type);
+        })
+        .catch((error: any) => {
+          console.log(error);
+          throw new Error(error);
+        });
+    }
+  }, [selectedNode]);
+
+  return (
+    <BPDrawer
+      className="drawer"
+      icon="properties"
+      title={title}
+      isOpen={isOpen}
+      onClose={onClose}
+      size={"30%"}
+    >
+      <div className={Classes.DRAWER_BODY}>
+        <div className={Classes.DIALOG_BODY}>
+          <H3 className="inline-title">
+            <span>Database credentials</span>
+            <span>{loading && <Spinner size={15} />}</span>
+          </H3>
+          <Mutation
+            mutation={upsertCredential}
+            onCompleted={(data: any) => {
+              setHasChanged(false);
+              setHasSuccessfullyChanged(true);
+            }}
+            onError={(error: any) => {
+              console.log("ERROR", error);
+            }}
+          >
+            {(upsert: any, { loading }: any) => {
+              return (
+                <form
+                  onSubmit={(e: any) => {
+                    console.log("onsubmit");
+                    e.preventDefault();
+                    upsert({
+                      variables: {
+                        host,
+                        port,
+                        login,
+                        database,
+                        password,
+                        type,
+                        sourceId: selectedNode.source.id
+                      }
+                    });
+                  }}
+                >
+                  <FormGroup label="Host">
+                    <InputGroup
+                      value={host}
+                      disabled={loading}
+                      leftIcon={"desktop"}
```

Comment:
where do these icons come from ? don't you have a react library where you can import them from ?

Reply:
They come from BlueprintJs, which ships with a set of icons!